src/shogun/lib/auto_init_factory.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef __AUTO_INIT_FACTORY_H__
+#define __AUTO_INIT_FACTORY_H__
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/kernel/Kernel.h>
+#include <shogun/lib/abstract_auto_init.h>
+#include <shogun/lib/any.h>
+
+namespace shogun
+{
+	namespace factory
+	{
+		class GammaFeatureNumberInit : public AutoInit
+		{
+			static const char* const kName;
+			static const char* const kDescription;
+
+		public:
+			explicit GammaFeatureNumberInit(CKernel* kernel)
+			    : AutoInit(kName, kDescription)
+			{
+				SG_REF(kernel);
+				m_kernel = kernel;
+			}
+
+			~GammaFeatureNumberInit() override
+			{
+				SG_UNREF(m_kernel);
+			}
+
+			Any operator()() override
+			{
+				Any result;
+				auto m_lhs = m_kernel->get<CFeatures*>("lhs");
```

Comment:
I think here you can use the actual getter for lhs (if it exists) as the kernel type will always be known. This way we know for sure there is no runtime error if the parameter is renamed or a typo happens or whatever?

Reply:
`CKernel::get_lhs`

Reply:
I must have missed it!